centreon/src/Core/ResourceAccess/Application/Repository/ReadResourceAccessRepositoryInterface.php

```diff
@@ -78,4 +79,44 @@ public function exists(int $ruleId): bool;
      * @return array<int, array<int>> [datasetId => [ResourceId1,ResourceId2, ...]]
      */
     public function findDatasetResourceIdsByHostGroupId(int $hostGroupId): array;
+
+    /**
+     * Return the list of rule ids that exist from the given rule IDs.
+     *
+     * @param int[] $ruleIds
+     * @return int[]
+     */
+    public function exist(array $ruleIds): array;
+
+    /**
+     * Check if a rule exists by contact.
+     * Existing rules are linked to the user ID or has "all_contacts" flag.
+     *
+     * @param int $userId
+     *
+     * @return int[]
+     */
+    public function existByContact(int $userId): array;
+
+    /**
+     * Check if a rule exists by contact groups.
+     * Existing rules are linked to the contact group IDs or has "all_contact_groups" flag.
+     *
+     * @param ContactGroup[] $contactGroups
+     *
+     * @return int[]
+     */
+    public function existByContactGroup(array $contactGroups): array;
```

Comment:
Those two methods do not respect the expected signature for existXXX() functions.
I would expect: 
existByContact(array $rulesIds, int $userId)
existByContactGroups(array $rulesIds, array $contactGroups)